App.tsx

```diff
@@ -1,27 +1,75 @@
 import { StatusBar } from 'expo-status-bar';
-import { StyleSheet, Text, View, Button, NativeModules } from 'react-native';
-import { useTranslation } from 'react-i18next'
-import './localization'
+import { StyleSheet, Text, TextInput, View, Button, NativeModules } from 'react-native';
+import { useTranslation } from 'react-i18next';
+import './localization';
+import { React, useState } from 'react';
+import * as SecureStore from 'expo-secure-store';
+
 
 const { CalendarModule, PrismModule } = NativeModules;
 
 export default function App() {
   const { t, i18n } = useTranslation()
-  
-  const onPress = () => {
-    console.log('DID: ' + PrismModule.createDID("passphrase"));
-  };
+
+  const [passphrase, setPassPhrase] = useState();
+
+  const [did, setDID] = useState('Your Generated DID');
+
+  const [securedids, setSecureDIDs] = useState('DIDs from passphrase');
 
   return (
     <View style={styles.container}>
       <Text>{t('Home.Title')}</Text>
-      <StatusBar style="auto" />
+      <View style={{ flexDirection: 'row' }}>
+      <Text>{t('Home.EnterPassPhrase')}:  </Text>
+      <TextInput
+        style={styles.textInput}
+        placeHolderText={passphrase}
+        clearTextOnFocus={true}
+        multiline={false}
+        onChangeText={text => setPassPhrase(text)}
+        value={passphrase}
+      />
       <Button
         title={t('Home.GenerateDIDButton')}
-        color="#841584"
-        onPress={onPress}
+        onPress={() => {
+          setDID(generateDID(passphrase));
+        }}
+      />
+      </View>
+      <View style={{ flexDirection: 'row' }}>
+      <Text>{t('Home.GeneratedDID')}:  </Text>
+      <TextInput
+        style={[styles.textInput,{width: 250}]}
+        maxLength={41}
+        clearTextOnFocus
+        multiline={false}
+        editable={false}
+        value={did}
+      />
+      <Button
+        title={t('Home.SavePassDID')}
+        onPress={() => {
+          savePassDID(passphrase, did);
```

Comment:
Are you using the key derivation binary seed passphrase also as a key for the SecureStore?